src/stash/src/postgres.rs

```diff
@@ -1019,3 +996,86 @@ impl Consolidator {
         Ok(())
     }
 }
+
+/// Stash factory to use for tests that allow creating multiple stashes.
+pub struct DebugStashFactory {
+    url: String,
+    schema: String,
+    tls: MakeTlsConnector,
+    stash_factory: StashFactory,
+}
+
+impl DebugStashFactory {
+    pub async fn new() -> Result<DebugStashFactory, StashError> {
+        let url =
+            std::env::var("COCKROACH_URL").expect("COCKROACH_URL environment variable is not set");
+        let rng: usize = rand::thread_rng().gen();
+        let schema = format!("schema_{rng}");
+        let tls = mz_postgres_util::make_tls(&tokio_postgres::Config::new()).unwrap();
+
+        let (client, connection) = tokio_postgres::connect(&url, tls.clone()).await?;
+        mz_ore::task::spawn(|| "tokio-postgres stash connection", async move {
+            if let Err(e) = connection.await {
+                tracing::error!("postgres stash connection error: {e}");
+            }
+        });
+        client
+            .batch_execute(&format!("CREATE SCHEMA {schema}"))
+            .await?;
+
+        let stash_factory = StashFactory::new(&MetricsRegistry::new());
+
+        Ok(DebugStashFactory {
+            url,
+            schema,
+            tls,
+            stash_factory,
+        })
+    }
+
+    pub async fn open_debug(&self) -> Result<Stash, StashError> {
+        self.stash_factory
+            .open(
+                self.url.clone(),
+                Some(self.schema.clone()),
+                self.tls.clone(),
+            )
+            .await
+    }
+}
+
+impl Drop for DebugStashFactory {
+    fn drop(&mut self) {
+        let url = self.url.clone();
+        let schema = self.schema.clone();
+        let tls = self.tls.clone();
+        let result = std::thread::spawn(move || {
+            let async_runtime = tokio::runtime::Builder::new_current_thread()
+                .enable_all()
+                .build()?;
+            async_runtime.block_on(async {
+                let (client, connection) = tokio_postgres::connect(&url, tls).await?;
+                mz_ore::task::spawn(|| "tokio-postgres stash connection", async move {
+                    if let Err(e) = connection.await {
+                        tracing::error!("postgres stash connection error: {e}");
+                    }
+                });
+                client
+                    .batch_execute(&format!("DROP SCHEMA {} CASCADE", &schema))
+                    .await?;
+                Ok::<_, StashError>(())
+            })
+        })
+        .join();
+
+        match result {
+            Ok(result) => {
+                if let Err(e) = result {
+                    tracing::error!("unable to cleanup debug stash: {e}");
+                }
+            }
+
+            Err(_) => tracing::error!("unable to cleanup debug stash"),
+        }
+    }
+}
```

Comment:
@doy-materialize and @guswynn I just added you two for this method. Feel free to ignore everything else.

Reply:
Also as an FYI, I tried to use the synchronous postgres crate that we use elsewhere, but it turns out those methods were all blocking on asynchronous methods. So I still got the same panic of 

```
        panic!(
                    "Cannot start a runtime from within a runtime. This happens \
                    because a function (like `block_on`) attempted to block the \
                    current thread while the thread is being used to drive \
                    asynchronous tasks."
                );
```